### DISCUSSION TITLE
How to add custom title to admonitions

### DISCUSSION THREAD

**Discussion by ma-sadeghi** [executablebooks/meta#834] [Q&A] [answered]
Is there a way to add a custom title to admonitions? For instance:

```
:::{note} My Custom Title!
**Wow**, a note!
:::
```

Thank you!

**Comment by NickleDave** [comment, marked as answer]
Not an expert but I think:

If you do this with the generic `admonition` directive then the text right after the directive will become the title. This is the way to add a custom title.
https://myst-parser.readthedocs.io/en/latest/syntax/roles-and-directives.html?highlight=admonition#admonitions
https://docutils.sourceforge.io/docs/ref/doctree.html#admonition

But for the `note` directive there's no arguments -- see how it's an example for how you can cram directives onto one line here.
https://myst-parser.readthedocs.io/en/latest/syntax/roles-and-directives.html#how-directives-parse-content
```
:::{note} Notes require **no** arguments, so content can start here.:::
```

That's because `note` is a more specific type of admonition that will always use its own name as the title, just like [caution](https://docutils.sourceforge.io/docs/ref/doctree.html#caution), [danger](https://docutils.sourceforge.io/docs/ref/doctree.html#danger), [error](https://docutils.sourceforge.io/docs/ref/doctree.html#error), [hint](https://docutils.sourceforge.io/docs/ref/doctree.html#hint), [important](https://docutils.sourceforge.io/docs/ref/doctree.html#important), [note](https://docutils.sourceforge.io/docs/ref/doctree.html#note), [tip](https://docutils.sourceforge.io/docs/ref/doctree.html#tip), [warning](https://docutils.sourceforge.io/docs/ref/doctree.html#warning)
https://docutils.sourceforge.io/docs/ref/doctree.html#attention

**Reply by ma-sadeghi** [reply]
Thank you so much.

**Reply by rowanc1** [reply]
This is certainly something that has confused a few people in the past as well, we are tracking this issue and will hopefully make some improvements:

* https://github.com/executablebooks/myst-spec/issues/49

Glad that you got it sorted and thanks @NickleDave for jumping in! 💚
